Molly/README.md

```diff
@@ -0,0 +1,168 @@
+### Homework #1
+(Due Monday 7/20)
+
+You need to turn in this homework by cloning this repo, making a new branch with your solutions, pushing to github, and issuing a pull request.
+For details, follow the instructions [here](http://portlandcodeschool.github.io/jse/2015/01/07/command-line-and-git-slides/#/14).
+
+Please also read the [learning objectives](objectives.md) for this week.
+
+---
+
+**1)** (_30% of total time_)
+Predict the output of each of the following expressions, and explain the rules which determine that answer.  (_Hint: The rules for operator `==` are complicated, but do your best!_)
+
+If the output depends on the value or type of variable _x_, identify the conditions (what _x_ is) when the expression will output true (or false, if that's simpler).  Assume the cases are independent, and _x_ is reset to an unknown value before each.
+
+Some of these are tricky!  Don't trust your first instinct.  
+
+
+**a)** `"1" == 1` True. When we convert "1" from a string into a number, 1 and 1 are equal.
+
+**b)** `"1" === 1` False. "1" is a string and 1 is a number. Therefore the two sides are not equal.
+
+**c)** `x == 'x'` False. x is undefined, which is always false-y
+
+**d)** `x == (x+'')` True. Both side are false, which makes them equal--thereby making this statement true.
+
+**e)** `'' == ' '` True. Both sides are false-y, which makes the statement true
+
+**f)** `x = true` True. We have assigned the value true to x, so the statement will come out as true.
+
+**g)** `var x; x == 'undefined'` False. x is undefined and therefore false-y while 'undefined' is a string and truth-y, which makes the two sides unequal.
+
+**h)** `'9'<'10'` True. The number 9 is less than the number 10. (I checked this in the console and know it's incorrect, but I can't figure out why.)
+
+**i)** `typeof x + 1 === "number"` False. The left side is undefined and the right side is a string.
+
+**j)** `typeof x % 2 === "number"` Fales. Left side is NaN and the right side is a string, so they are not the same.
+
+**k)** `typeof (x % 2) === "number"` True, but I can't explain why. Somehow the parenthesis fixes things.
+
+**l)** `x++ == ++x` False, but I can't explain why.
+
+**m)** `++x == x++` True, but I can't explain why.
+
```

Comment:
In a similar situation to the above, because of the way that ++x and x++ work, we are returned values in a different order.

++x will add 1 to the value of x prior to returning it to the equation, so we end up (assuming x = 1) with 2 on the left side.

x++ will return the value before adding anything to it, so even though it is going to add 1 to x, it actually gives the equation back a 2 before it does that, resulting in 2 == 2, which is true.